### ISSUE TITLE
When testing using FakeRedis, SimpleWorker crashes if connection not explicitly passed

### ISSUE THREAD

**Issue by jamescooke** [nvie/rq#648]
Working on master in order to use FakeRedis and https://github.com/nvie/rq/pull/619 , I've found that `SimpleWorker` will crash if it's not passed a connection explicitly.

This is my failing test (following the strategy on http://python-rq.org/docs/testing/):

        from fakeredis import FakeStrictRedis
        from rq import SimpleWorker, Queue

        fake_redis = FakeStrictRedis()
        queue = Queue(connection=fake_redis)
        worker = SimpleWorker([queue])

With stack trace of:

```
Traceback (most recent call last):
  File "/home/james/test_queueing_test_case.py", line 22, in test_a
    worker = SimpleWorker([queue])
  File "/home/james/venv/src/rq/rq/worker.py", line 151, in __init__
    self.failed_queue = get_failed_queue(connection=self.connection)
  File "/home/james/venv/src/rq/rq/queue.py", line 20, in get_failed_queue
    return FailedQueue(connection=connection)
  File "/home/james/venv/src/rq/rq/queue.py", line 425, in __init__
    super(FailedQueue, self).__init__(JobStatus.FAILED, connection=connection)
  File "/home/james/venv/src/rq/rq/queue.py", line 59, in __init__
    self.connection = resolve_connection(connection)
  File "/home/james/venv/src/rq/rq/connections.py", line 70, in resolve_connection
    raise NoRedisConnectionException('Could not resolve a Redis connection')
NoRedisConnectionException: Could not resolve a Redis connection
```

The solution appears to be to pass `connection=fake_redis` to the worker as follows:

    worker = SimpleWorker([queue], connection=fake_redis)

... which passes and is able to consume tasks. So it's not a blocking issue (it's great to be able to test using FakeRedis :+1:), but it might trip up others, and / or, points to some issue in `resolve_connection`?

**Comment by nvie** [comment]
Fixed by #693

**Comment by jamescooke** [comment]
Awesome :+1: Thanks
